Update Documentation for 0.7.2

HylianMedia opened the pull request:
I have updated various sections of the documentation to reflect Atmosphère 0.7.2.

A couple of key changes I would like to mention:
The release roadmap is now a list of upcoming features. I have done this because as far as I am aware, we do not yet know in what version certain features will be added. When a feature on this list is added, I will remove it from the list.

I have added a changelog to the documentation. This log contains release notes for all versions of Atmosphère.

Finally, I have updated the HBL section of loader.md to explain how we are able to use any game on the Home Menu to launch hbmenu, as opposed to launching it from the album.

yellows8 commented on docs/modules/loader.md:
```diff
@@ -53,7 +53,13 @@ When authoring patches, [hactool](https://github.com/SciresM/hactool) can be use
 ### HBL Support
 
-TODO
+Atmosphère can use the loader module in order to turn any game on your Switch's home menu into a launchpoint for the Homebrew Menu, rather than launching it through the album applet. This allows one to launch the Homebrew Menu with access to all 4GB of RAM on the Switch, as opposed to the extremely small portion of RAM we are limited to when launching the Homebrew Menu from the album. This also m ...
```
"4GB" No, see nx-hbloader release notes.

HylianMedia commented:
Quick update, fixed some mistakes pointed out by @yellows8 and moved the information regarding button overrides to the Button Overrides section in loader.md.

hexkyz commented:
LGTM
